### ISSUE TITLE
stdout/stderr should not be line buffered 

### ISSUE THREAD

**Issue by ry** [denoland/deno#948]
we are using [tokio-fs's stdout](https://docs.rs/tokio-fs/0.1.0/tokio_fs/struct.Stdout.html), which writes to [std::io::Stdout](https://doc.rust-lang.org/std/io/struct.Stdout.html)

Rust's stdout is always line buffered: 
https://github.com/rust-lang/rust/blob/e1643a8968753226dab7ab3c9fee826f097550f2/src/libstd/io/stdio.rs#L344-L349
We are not so happy with this behavior and likely want an always unbuffered stdout, as in Node. To support his behavior we will need to refactor the stdout/stderr resources to use a raw file descriptor like this:
```rust
use std::os::unix::io::FromRawFd;
let stdout = File::from_raw_fd(1);
```

this came up in https://github.com/denoland/deno/pull/946

**Comment by hayd** [comment]
Is an alternative to flush after each write? Or is the concern that that'd be inefficient?

**Comment by ry** [comment]
I think line buffered is confusing - and especially so when we don’t even allow directly writing strings - so I’d prefer to not add flush

**Comment by hayd** [comment]
I mean, flush in rust after each write: https://github.com/hayd/deno/commit/db4ca6cce40d0823423dab35abdd96b934149e31

**Comment by ry** [comment]
@hayd I'd rather not call flush. It seems excessive.

**Comment by hayd** [comment]
Your PR `fs::File::from_std()` was included in into tokio-fs-0.1.4 (which deno is using now).

**Comment by ry** [comment]
I got half way done with this before getting distracted. Will try to fix up in the next few days.
WIP here https://github.com/ry/deno/commit/756078733185e9127901c6957a73d0e2c9bdafde
